app/controllers/users/devices_controller.rb

```diff
@@ -23,14 +25,14 @@ def new
   end
 
   def shared
-    @presenter = ::Users::DevicesPresenter.new(current_user, params, 'shared')
-    gon.push(@presenter.shared_gon)
+    @devices_shared_presenter = ::Users::Devices::DevicesSharedPresenter.new(current_user, params)
+    gon.push(@devices_shared_presenter.shared_gon)
   end
 
   def info
-    presenter = ::Users::DevicesPresenter.new(current_user, params, 'info')
-    @device = presenter.device
-    @config = presenter.config
+    @devices_info_presenter = ::Users::Devices::DevicesInfoPresenter.new(current_user, params)
+    @device = @devices_info_presenter.device
```

Comment:
there is no much sense in presenter if you have to instantiate other variables in controller